lib/src/plugin_core/reaction.rs

```diff
@@ -0,0 +1,190 @@
+///
+/// pub struct MyReaction {
+///     base: ReactionBase,
+///     // Plugin-specific fields
+/// }
+///
+/// impl MyReaction {
+///     // No event_tx needed - it's injected automatically by DrasiLib
+///     pub fn new(config: MyReactionConfig) -> Self {
+///         let params = ReactionBaseParams::new(&config.id, config.queries.clone())
+///             .with_priority_queue_capacity(config.queue_capacity);
+///
+///         Self {
+///             base: ReactionBase::new(params),
+///         }
+///     }
+/// }
+///
+/// #[async_trait]
+/// impl Reaction for MyReaction {
+///     fn id(&self) -> &str {
+///         &self.base.id
+///     }
+///
+///     fn type_name(&self) -> &str {
+///         "my-reaction"
+///     }
+///
+///     fn query_ids(&self) -> Vec<String> {
+///         self.base.queries.clone()
+///     }
+///
+///     async fn inject_event_tx(&self, tx: ComponentEventSender) {
+///         self.base.inject_event_tx(tx).await;
+///     }
+///
+///     // ... implement other methods
+/// }
+/// ```
+#[async_trait]
+pub trait Reaction: Send + Sync {
+    /// Get the reaction's unique identifier
+    fn id(&self) -> &str;
+
+    /// Get the reaction type name (e.g., "http", "log", "sse")
+    fn type_name(&self) -> &str;
+
+    /// Get the reaction's configuration properties for inspection
+    ///
+    /// This returns a HashMap representation of the reaction's configuration
+    /// for use in APIs and inspection. The actual typed configuration is
+    /// owned by the plugin - this is just for external visibility.
+    fn properties(&self) -> std::collections::HashMap<String, serde_json::Value>;
+
+    /// Get the list of query IDs this reaction subscribes to
+    fn query_ids(&self) -> Vec<String>;
+
+    /// Start the reaction with access to query subscriptions
+    ///
+    /// The reaction should:
+    /// 1. Subscribe to all configured queries via query_subscriber
+    /// 2. Start its processing loop
+    /// 3. Update its status to Running
+    async fn start(&self, query_subscriber: Arc<dyn QuerySubscriber>) -> Result<()>;
+
+    /// Stop the reaction, cleaning up all subscriptions and tasks
+    async fn stop(&self) -> Result<()>;
+
+    /// Get the current status of the reaction
+    async fn status(&self) -> ComponentStatus;
+
+    /// Inject the event channel for component lifecycle events
+    ///
+    /// This method is called automatically by DrasiLib when the reaction is added
+    /// via `add_reaction()`. Plugin developers do not need to call this directly.
+    ///
+    /// Implementation should delegate to `self.base.inject_event_tx(tx).await`.
+    async fn inject_event_tx(&self, tx: ComponentEventSender);
+}
+
+/// Blanket implementation of Reaction for Box<dyn Reaction>
+///
+/// This allows boxed trait objects to be used with methods expecting `impl Reaction`.
+#[async_trait]
+impl Reaction for Box<dyn Reaction + 'static> {
```

Comment:
Do we want reactions to live for the entire lifespan of the application? Why do we need `'static`?